src/components/Docs/components/AddressConverter/components/AddressConverter.tsx

```diff
@@ -1,40 +1,163 @@
+        {address.startsWith("0x") && isValidAddress(address) && (
+          <p>
+            {isChecksumAddress ? (
+              "This address is checksummed."
+            ) : (
+              <>
+                This address is{" "}
+                <a
+                  href="https://docs.alchemy.com/docs/how-to-handle-checksum-addresses"
+                  target="_blank"
+                  rel="noopener noreferrer"
+                  className="text-blue-500 underline"
+                >
+                  not checksummed
+                </a>
+                .
+              </>
+            )}
+          </p>
+        )}
+      </div>
+
       <div className="flex items-center gap-3 mb-3">
         <input
           type="text"
           value={address}
-          onChange={(event) => setAddress(event.target.value)}
+          onChange={handleAddressChange}
           placeholder="0x or zeta address"
-          className="border border-grey-200 dark:border-grey-500 rounded p-3 sm:w-[450px] bg-transparent dark:bg-grey-800 outline-none"
+          autoComplete="off"
+          className="tracking-wide border border-gray-200 dark:border-gray-500 rounded p-3 sm:w-[450px] bg-transparent dark:bg-gray-800 outline-none"
         />
-
         <button
           type="button"
-          onClick={() => {
-            try {
-              const converted = convertAddress(address);
-
-              setConvertedAddress(converted);
-            } catch (error) {
-              console.error(error);
-            }
-          }}
-          className="border border-grey-200 dark:border-grey-500 rounded p-3 hover:border-green-100 hover:dark:border-green-100 transition-[border-color]"
+          onClick={handleConversion}
+          disabled={!address || !isValidAddress(address)}
+          className="border border-gray-200 dark:border-gray-500 rounded p-3 hover:border-green-100 hover:dark:border-green-100 transition disabled:opacity-50 disabled:cursor-not-allowed min-w-[150px] whitespace-nowrap"
         >
-          Convert
+          {getButtonStatus()}
         </button>
       </div>
 
-      <p style={{ marginBottom: "1rem" }}>Converted address: {convertedAddress}</p>
+      {errorMessage && <p className="text-red-500 mb-3">{errorMessage}</p>}
+
+      {convertedAddress && (
+        <div className="flex items-center gap-3 mb-3">
+          <input
+            type="text"
+            value={convertedAddress}
+            disabled
+            className="tracking-wide border border-gray-200 dark:border-gray-500 rounded p-3 sm:w-[450px] bg-transparent dark:bg-gray-800 outline-none"
+          />
+          <button
+            onClick={copyToClipboard}
+            className="border border-gray-200 dark:border-gray-500 rounded p-3 hover:border-green-100 hover:dark:border-green-100 transition min-w-[150px] whitespace-nowrap"
+          >
+            Copy
+          </button>
+        </div>
+      )}
+
+      {convertedAddress && (
```

Comment:
This is the same check as above, can be in within the same check, maybe wrapping it within a fragment?

Reply:
69cbbb6